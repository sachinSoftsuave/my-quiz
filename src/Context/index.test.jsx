import React from "react";
import "jsdom-global/register";
import Enzyme, { mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Main from ".";
import { QuizContext } from '.';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

let mockSetQuizStarted = jest.fn();
let mockSetMinutes = jest.fn();
let mockIsActive = jest.fn();

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useContext: () => ({
    isQuizStarted: false,
    isActive: false,
    minutes: false,
    setQuizStarted: () => mockSetQuizStarted,
    setMinutes: () => mockSetMinutes,
    setIsActive: () => mockIsActive
  })

}));

 const value = {
     mockIsActive,
     mockSetMinutes,
     mockSetQuizStarted
  };

  describe("Context", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Main />);
  });

  it("should render the snapshot of main component", () => {
  const wrapper = mount(<QuizContext.Provider value={value}><Main /></QuizContext.Provider>);
  expect(toJson(wrapper)).toMatchSnapshot();
  });
  
  })


