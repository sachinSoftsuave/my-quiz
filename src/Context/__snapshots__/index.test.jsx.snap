// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context should render the snapshot of main component 1`] = `<Main />`;
